src/js/core/methods/tx/refTx.js

```diff
@@ -202,3 +204,88 @@ export const transformReferencedTransactions = (
             },
         ];
     });
+
+// Validate referenced transactions provided by the user.
+// Data sent as response to TxAck needs to be strict.
+// They should not contain any fields unknown/unexpected by protobuf.
+export const validateReferencedTransactions = (
+    txs: RefTransaction[] | typeof undefined,
+    inputs: TxInputType[],
+    outputs: TxOutputType[],
+) => {
+    if (!Array.isArray(txs) || !txs.length) return; // allow empty, they will be downloaded later...
```

Comment:
I don't know what is our convention but I generally preffer explicit booleans in conditionals. RBY.

```suggestion
    if (!Array.isArray(txs) || txs.length === 0) return; // allow empty, they will be downloaded later...
```

Reply:
c6b5906

Reply:
🦊